packages/network/test/integration/browser-ws-endpoint.test.ts

```diff
@@ -93,4 +93,23 @@ describe('ws-endpoint', () => {
             expect(close[1]).toContain('Handshake not received from connection behind UUID')
         })
     })
+
+    describe('Duplicate connections from same nodeId are closed', () => {
+        it('Duplicate connection is closed', async () => {
+            const client1 = new BrowserClientWsEndpoint(PeerInfo.newNode('client'))
+            const client2 = new BrowserClientWsEndpoint(PeerInfo.newNode('client'))
+
+            const server = await startServerWsEndpoint('127.0.0.1', 38482, PeerInfo.newNode('server'))
+
+            await client1.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            try {
+                await client2.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            } catch (err) {
+                expect(err).toContain('private key')
+            }
+            await client1.stop()
+            await client2.stop()
+            await server.stop()
```

Comment:
Probably broke indentation here, but to ensure cleanup happens: try-finally.
```suggestion
            try {
              await client1.connect(server.getUrl(), PeerInfo.newTracker('server'))
              try {
                  await client2.connect(server.getUrl(), PeerInfo.newTracker('server'))
              } catch (err) {
                  expect(err).toContain('private key')
              }
          } finally {
              await client1.stop()
              await client2.stop()
              await server.stop()
          }
```